zellij-server/src/panes/tiled_panes/mod.rs

```diff
@@ -917,7 +918,35 @@ impl TiledPanes {
                     *self.display_area.borrow(),
                     *self.viewport.borrow(),
                 );
-                let next_index = pane_grid.next_selectable_pane_id_to_the_left(&active_pane_id);
+                let mut next_index = pane_grid.next_selectable_pane_id_to_the_left(&active_pane_id);
+
+                if next_index.is_none() && wrap_around {
+                    let max_x = self
+                        .panes
+                        .iter()
+                        .filter(|(_, pane)| pane.selectable())
+                        .max_by_key(|(_, pane)| pane.x())
+                        .map(|(_, pane)| pane.x());
+
+                    next_index = if let Some(max_x) = max_x {
+                        let candidates = self
+                            .panes
+                            .values()
+                            .filter(|pane| pane.x() == max_x && pane.selectable())
```

Comment:
I think this is not accurate. Shouldn't it also include the pane width?

Reply:
IMHO, the rightmost selectable pane  = pane.x is the larger number, the width does not really matter here 

I've tested a lot of times, and this solution works fine.

e.g - 
A(current focus) | B | C
-- | -- | --
D | E | F

When I `alt + <-` to move left, I just need to locate the pane with maximum `x` && `selectable = true`, if there are multiple, then the second part of logic comes for the help which is to calculate the `abs` of y -> Pane `C`

Did I miss a scenario that must consider the width?

Reply:
@imsnif  Sorry for the late reply, i was sick in last a few days. I've made a few changes, now the logic follows:

1. It filters the panes to find the ones with the maximum value of the x-coordinate plus the pane's width.
2. If there are multiple candidates, it further filters the candidates to find the panes with the maximum x-coordinate.
3. If there are still multiple candidates, it selects the one with the minimum gap in the y-coordinate.

Reply:
> Looking much better! I left some comments about the implementation. Also - is there any reason we're only doing right+left and not also up+down?

I added up + down as well